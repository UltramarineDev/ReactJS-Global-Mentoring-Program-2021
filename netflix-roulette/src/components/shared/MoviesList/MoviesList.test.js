import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import MovieList from './MoviesList';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
  useSelector: jest.fn().mockReturnValue({ movies: {}, search: '', isMoviesLoading: false, isMovieDeleted: false }),
}));

describe('MovieList tests', () => {
  test('renders MovieList', () => {
    const { asFragment } = render(<MovieList />);
    expect(asFragment(<MovieList />)).toMatchSnapshot();
  });
});
