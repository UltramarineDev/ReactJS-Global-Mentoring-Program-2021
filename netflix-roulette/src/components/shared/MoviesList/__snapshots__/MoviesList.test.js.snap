// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList tests renders MovieList 1`] = `
<DocumentFragment>
  <div
    class="movieListSection"
  >
    <div
      class="navBar"
    >
      <div
        class="resultsFilter"
      >
        <a
          class="tab activeTab"
        >
          ALL
        </a>
        <a
          class="tab"
        >
          DOCUMENTARY
        </a>
        <a
          class="tab"
        >
          COMEDY
        </a>
        <a
          class="tab"
        >
          HORROR
        </a>
        <a
          class="tab"
        >
          CRIME
        </a>
      </div>
      <div
        class="resultsSort"
      >
        <span>
          SORT BY
        </span>
        <div
          class="dropdown"
        >
          <a
            aria-label="set_options_dropdown_label"
            id="set_options_dropdown"
          >
            RELEASE DATE
          </a>
          <div
            class="icon"
          />
        </div>
      </div>
    </div>
    <div
      class="border"
    />
    <div
      class="moviesCount"
    >
      <strong>
         
      </strong>
      movies found
    </div>
  </div>
</DocumentFragment>
`;
